api/types/kubewaitingcontainer/kubewaitingcontainer.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2024 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubewaitingcontainer
+
+import (
+	"time"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/header"
+	"github.com/gravitational/teleport/api/types/header/convert/legacy"
+)
+
+// KubeWaitingContainer is a Kubernetes ephemeral
+// container that are waiting to be created until moderated
+// session conditions are met.
+type KubeWaitingContainer struct {
```

Comment:
The proto message doesn't implement `api/types.Resource` so it can't be used bare in `api/client.EventToGRPC` for example. I think there's a few other places were a `types.Resource` is required as well.

Reply:
Thanks, removed the pure Go copy